src/emoji/Emoji.js

```diff
@@ -1,9 +1,37 @@
 /* @flow strict-local */
-import { createIconSet } from 'react-native-vector-icons';
-import { nameToEmojiMap } from './data';
+import React, { PureComponent } from 'react';
 
-/* $FlowFixMe: `nameToEmojiMap` is mistyped upstream; elements of
-  `glyphMap` may be either `number` or `string`. */
-const Icon = createIconSet(nameToEmojiMap, 'AppleColorEmoji');
+import type { ImageEmojiType, Dispatch } from '../types';
+import { connect } from '../react-redux';
+import UnicodeEmoji from './UnicodeEmoji';
+import ImageEmoji from './ImageEmoji';
+import { getActiveImageEmojiByName } from './emojiSelectors';
 
-export default Icon;
+type SelectorProps = {|
+  imageEmoji: ImageEmojiType | void,
+|};
+
+type Props = {|
+  name: string,
+
+  dispatch: Dispatch,
+  ...SelectorProps,
+|};
+
+type ComponentProps = {|
+  name: string,
+|};
+
+class Emoji extends PureComponent<Props> {
+  render() {
+    const { name, imageEmoji } = this.props;
+
+    // TODO: this only handles Unicode emoji (shipped with the app)
+    // and realm emoji, but not Zulip extra emoji.  See our issue #2846.
+    return imageEmoji ? <ImageEmoji emoji={imageEmoji} /> : <UnicodeEmoji name={name} size={20} />;
+  }
+}
+
+export default connect((state, props: ComponentProps): SelectorProps => ({
+  imageEmoji: getActiveImageEmojiByName(state)[props.name],
```

Comment:
Is this still the right set of emoji? For an old reaction, there might be an inactive emoji, I think.

(Any change should be a separate commit from the refactor that moves this code, though.)

Reply:
yup, let's take it in another commit.

Let's do it in follow up PR.
Because `<Emoji` is used in `<EmojiRow`. And `<EmojiRow` is used in `EmojiPIckerScreen` and `EmojiAutocomplete`, where we only wants active emojis.

Reply:
Added c1058092